MonitoringSystem/Service/MonitoringHandler.py

```diff
@@ -0,0 +1,263 @@
+    if not isinstance( reportRequest[ 'extraArgs' ], self.__reportRequestDict[ 'extraArgs' ] ):
+      return S_ERROR( "Extra args has to be of type %s" % self.__reportRequestDict[ 'extraArgs' ] )
+    reportRequestExtra = reportRequest[ 'extraArgs' ]
+    
+    # Check sliding plots
+    if 'lastSeconds' in reportRequestExtra:
+      try:
+        lastSeconds = long( reportRequestExtra[ 'lastSeconds' ] )
+      except ValueError:
+        gLogger.error( "lastSeconds key must be a number" )
+        return S_ERROR( "Value Error" )
+      #TODO: Maybe we can have last hour in the monitoring
+      if lastSeconds < 3600:
+        return S_ERROR( "lastSeconds must be more than 3600" )
+      now = Time.dateTime()
+      reportRequest[ 'endTime' ] = now
+      reportRequest[ 'startTime' ] = now - datetime.timedelta( seconds = lastSeconds )
+    else:
+      # if end date is not there, just set it to now
+      if not reportRequest.get( 'endTime', False ):
+        reportRequest[ 'endTime' ] = Time.dateTime()
+    # Check keys
+    for key in self.__reportRequestDict:
+      if not key in reportRequest:
+        return S_ERROR( 'Missing mandatory field %s in plot reques' % key )
+
+      if not isinstance( reportRequest[ key ], self.__reportRequestDict[ key ] ):
+        return S_ERROR( "Type mismatch for field %s (%s), required one of %s" % ( key,
+                                                                                  str( type( reportRequest[ key ] ) ),
+                                                                                  str( self.__reportRequestDict[ key ] ) ) )
+      if key in ( 'startTime', 'endTime' ):
+        reportRequest[ key ] = int( Time.toEpoch( reportRequest[ key ] ) )
+    
+    return S_OK( reportRequest )
+
+  types_generatePlot = [ types.DictType ]
+  def export_generatePlot( self, reportRequest ):
+    """
+    It crated a plots for a given request
+    :param dict reportRequest contains the plot arguments...
+    """
+    retVal = self.__checkPlotRequest( reportRequest )
+    if not retVal[ 'OK' ]:
+      return retVal
+    reporter = MainReporter( self.__db, self.serviceInfoDict[ 'clientSetup' ] )
+    reportRequest[ 'generatePlot' ] = True
+    return reporter.generate( reportRequest, self.getRemoteCredentials() )
+  
+  types_getReport = [ types.DictType ]
+  def export_getReport( self, reportRequest ):
+    """
+    It is used to get the raw data used to create a plot.
+    :param str typeName the type of the monitoring
+    :param str reportName the name of the plotter used to create the plot for example:  NumberOfJobs
+    :param int startTime epoch time, start time of the plot
+    :param int endTime epoch time, end time of the plot
+    :param dict condDict is the conditions used to gnerate the plot: {'Status':['Running'],'grouping': ['Site'] }
+    :param str grouping is the grouping of the data for example: 'Site'
+    :paran dict extraArgs epoch time which can be last day, last week, last month
```

Comment:
typo: param
Also needs closing colons, also newline before the list of ":params"